openzwave-sys/openzwave-c/value_classes/value_id.cc

```diff
@@ -4,60 +4,69 @@
 extern "C" {
 
 // constructors
-ValueID * value_id_from_packed_id(uint32 home_id, uint64 id) {
-  return new ValueID(home_id, id);
+cValueID value_id_from_packed_id(uint32 home_id, uint64 id) {
+  cValueID vid(home_id, id);
+  return vid;
 }
-
-ValueID * value_id_from_values(uint32 home_id,
-                               uint8 node_id,
-                               ValueGenre genre,
-                               uint8 command_class_id,
-                               uint8 instance,
-                               uint8 value_index,
-                               ValueType type) {
-  return new ValueID(home_id, node_id, genre, command_class_id, instance, value_index, type);
+cValueID value_id_from_values(uint32 home_id,
+                            uint8 node_id,
+                            ValueGenre genre,
+                            uint8 command_class_id,
+                            uint8 instance,
+                            uint8 value_index,
+                            ValueType type) {
+  ValueID vid(home_id, node_id, genre, command_class_id, instance, value_index, type);
+  cValueID cvid(vid.GetHomeId(), vid.GetId());
+  return cvid;
 }
 
 // instance methods
-uint32 value_id_get_home_id(ValueID * value) {
-  return value->GetHomeId();
-}
-
-uint8 value_id_get_node_id(ValueID * value) {
-  return value->GetNodeId();
+uint8 value_id_get_node_id(cValueID *cvid) {
+  ValueID vid(cvid->home_id, cvid->id);
```

Comment:
I think that we can do a cleanup and get rid of cValueID and just use ValueID here.

What's important is that the rust side
```
#[repr(C)]
pub struct ValueID {
  pub home_id: uint32_t,
  pub id: uint64_t
}
```
match the C++ struct/class. So we could change the above (from openzwave-sys/src/value_classes/value_id.rs) to be:
```
#[repr(C)]
pub struct ValueID {
  pub id: uint32_t,
  pub id1: uint32_t,
  pub home_did: uint32_t
}
```
and then that ValueID would match the C++ ValueID and we could eliminate cValueID (since the C++ ValudID class has no base classes and no virtual members, it's essentially the same thing as a plain old struct).

But I think I'd leave that for a separate PR since what we have now works, even if it isn't quite as efficient as it could be.